crates/hir/src/lib.rs

```diff
@@ -6343,3 +6352,42 @@ where
         self(item)
     }
 }
+
+pub fn resolve_absolute_path(
+    db: &dyn HirDatabase,
+    mut segments: impl Iterator<Item = Symbol>,
+) -> impl Iterator<Item = ItemInNs> {
+    segments
+        .next()
+        .and_then(|crate_name| {
+            let krate = *db.all_crates().iter().find(|&krate| {
+                krate
+                    .extra_data(db)
+                    .display_name
+                    .as_ref()
+                    .is_some_and(|name| *name.crate_name().symbol() == crate_name)
+            })?;
+            let mut def_map = db.crate_def_map(krate);
+            let mut module = &def_map[DefMap::ROOT];
+            let mut segments = segments.with_position().peekable();
+            while let Some((_, segment)) = segments.next_if(|&(position, _)| {
+                !matches!(position, itertools::Position::Last | itertools::Position::Only)
+            }) {
+                let res = module
+                    .scope
+                    .get_including_legacy_macros(&Name::new_symbol_root(segment))
```

Comment:
It feels wrong to me that we consider legacy scoped macros. These are never accessible outside their own crate anyways (if they are re-exported they won't be treated as legacy macros anymore but as proper item scoped macros). So this handling looks unnecssary ot me